networkdb/cluster.go

```diff
@@ -44,6 +44,8 @@ func (l *logWriter) Write(p []byte) (int, error) {
 
 // SetKey adds a new key to the key ring
 func (nDB *NetworkDB) SetKey(key []byte) {
+	nDB.Lock()
+	defer nDB.Unlock()
 	logrus.Debugf("Adding key %s", hex.EncodeToString(key)[0:5])
```

Comment:
That is not the purpose of lock. After printing the debug if the lock can't be acquired the go routine is going to be stuck. If its a deadlock the stack decode will capture the state when the user dumps it.